org.eclipse.scanning.points/src/org/eclipse/scanning/jython/JythonInterpreterManager.java

```diff
@@ -0,0 +1,240 @@
+	    	ne.printStackTrace();
+	    	logger.debug("Problem loading jython bundles!", ne);
+	    }
+	}
+
+
+	private static ClassLoader createJythonClassLoader(ClassLoader classLoader) {
+		
+    	ClassLoader jythonClassloader = ScanPointGeneratorFactory.class.getClassLoader();
+    	
+    	try { // For non-unit tests, attempt to use the OSGi classloader of this bundle.
+    		String jythonBundleName = System.getProperty("org.eclipse.scanning.jython.osgi.bundle.name", "uk.ac.diamond.jython");
+    		CompositeClassLoader composite = new CompositeClassLoader(classLoader);
+   	  	    // Classloader for org.eclipse.scanning.points
+    		composite.addLast(ScanPointGeneratorFactory.class.getClassLoader());
+    		addLast(composite, jythonBundleName);
+    		jythonClassloader = composite;
+     		
+    	} catch (Throwable ne) {
+    		ne.printStackTrace();
+    		// Legal, if static classloader does not work in tests, there will be
+    		// errors. If bundle classloader does not work in product, there will be errors.
+    		// Typically the message is something like: 'cannot find module org.eclipse.scanning.api'
+    	}
+    	return jythonClassloader;
+	}
+
+	private static void setJythonPaths(PySystemState state) {
+    	        	
+        try {
+        	// Search for the Libs directory which should have been expanded out either
+        	// directly into the bundle or into the 'jython2.7' folder.
+	    	String jythonBundleName = System.getProperty("org.eclipse.scanning.jython.osgi.bundle.name", "uk.ac.diamond.jython");
+
+            File lib = getBundleLocation(jythonBundleName); // TODO Name the jython OSGi bundle without Diamond in it!
+	    	Iterator<Object> it = state.path.iterator();
+	    	while (it.hasNext()) {
+	    		Object ob = it.next();
+	    		if (ob instanceof String && ((String)ob).endsWith(File.separator+"Lib")) {
+	    			lib = new File((String)ob);
+	    		}
+	    	}
+	    	
+	    	if (lib==null) {
+	            File loc = getBundleLocation(jythonBundleName); // TODO Name the jython OSGi bundle without Diamond in it!
+           	    File jythonDir = find(loc, "jython");
+            	lib       = find(jythonDir, "Lib");
+	    	}
+	    	
+            if (lib != null) {
+            	File site       = find(lib, "site-packages");
+            	state.path.add(new PyString(site.getAbsolutePath())); // Resolves the collections
+
+            	File[] fa = site.listFiles();
+            	for (File dir : fa) {
+            		if (!dir.isDirectory()) continue;
+            		if (dir.getName().endsWith("-info")) continue;
+            		state.path.add(new PyString(dir.getAbsolutePath())); // Resolves the collections
+            	}
+            	
+            	state.toString();
+
+            	System.out.println(state.path);
+            }
+            
+        } catch (Exception ne) {
+        	System.out.println("Problem setting jython path to include scripts");
+        	ne.printStackTrace();
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Use a logger to log this exception. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1148)